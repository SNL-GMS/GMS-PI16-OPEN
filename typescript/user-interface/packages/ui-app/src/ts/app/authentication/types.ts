import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
