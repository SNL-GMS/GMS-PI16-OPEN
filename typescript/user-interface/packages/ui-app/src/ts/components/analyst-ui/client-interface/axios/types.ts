/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
