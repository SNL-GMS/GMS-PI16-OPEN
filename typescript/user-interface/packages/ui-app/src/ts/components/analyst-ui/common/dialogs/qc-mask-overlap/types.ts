import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
