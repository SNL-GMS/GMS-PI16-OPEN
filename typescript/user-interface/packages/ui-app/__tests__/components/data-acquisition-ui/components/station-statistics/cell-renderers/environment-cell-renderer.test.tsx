import React from 'react';

import {
  ChannelEnvironmentCellRenderer,
  StationEnvironmentCellRenderer
} from '../../../../../../src/ts/components/data-acquisition-ui/components/station-statistics/cell-renderers/environment-cell-renderer';
import { StationStatisticsTableDataContext } from '../../../../../../src/ts/components/data-acquisition-ui/components/station-statistics/types';
import { cellRendererProps, row } from './common-test-data';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

describe('Environment cell renderer', () => {
  const channelEnvironmentCellRenderer = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <StationStatisticsTableDataContext.Provider value={{ data: [row] }}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <ChannelEnvironmentCellRenderer {...cellRendererProps} />
    </StationStatisticsTableDataContext.Provider>
  );

  const stationEnvironmentCellRenderer = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <StationStatisticsTableDataContext.Provider value={{ data: [row] }}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <StationEnvironmentCellRenderer {...cellRendererProps} />
    </StationStatisticsTableDataContext.Provider>
  );

  const nullStationEnvironmentCellRenderer = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <StationStatisticsTableDataContext.Provider value={{ data: [] }}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <StationEnvironmentCellRenderer {...cellRendererProps} />
    </StationStatisticsTableDataContext.Provider>
  );

  it('ChannelEnvironmentCellRenderer can be created', () => {
    expect(channelEnvironmentCellRenderer).toMatchSnapshot();
  });

  it('StationEnvironmentCellRenderer can be created', () => {
    expect(stationEnvironmentCellRenderer).toMatchSnapshot();
  });

  it('can be created handle no data', () => {
    expect(nullStationEnvironmentCellRenderer).toMatchSnapshot();
  });
});
