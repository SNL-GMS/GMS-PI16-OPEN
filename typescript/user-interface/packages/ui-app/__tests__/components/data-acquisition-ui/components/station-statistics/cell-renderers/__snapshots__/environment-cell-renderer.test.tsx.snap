// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment cell renderer ChannelEnvironmentCellRenderer can be created 1`] = `
<ChannelEnvironmentCellRenderer
  context={
    Object {
      "data": Object {
        "channelEnvironment": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelLag": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelMissing": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelTimeliness": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "id": "test",
        "needsAcknowledgement": false,
        "needsAttention": true,
        "stationData": Object {
          "stationCapabilityStatus": "GOOD",
          "stationName": "test",
          "stationStatus": "GOOD",
        },
        "stationEnvironment": 1,
        "stationGroups": Array [
          Object {
            "groupName": "Group",
            "sohStationCapability": "GOOD",
            "stationName": "test",
          },
        ],
        "stationLag": 2,
        "stationMissing": 3,
        "stationTimeliness": 4,
      },
    }
  }
  data={
    Object {
      "channelEnvironment": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelLag": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelMissing": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelTimeliness": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "id": "test",
      "needsAcknowledgement": false,
      "needsAttention": true,
      "stationData": Object {
        "stationCapabilityStatus": "GOOD",
        "stationName": "test",
        "stationStatus": "GOOD",
      },
      "stationEnvironment": 1,
      "stationGroups": Array [
        Object {
          "groupName": "Group",
          "sohStationCapability": "GOOD",
          "stationName": "test",
        },
      ],
      "stationLag": 2,
      "stationMissing": 3,
      "stationTimeliness": 4,
    }
  }
  formatValue={[MockFunction]}
  getValue={[MockFunction]}
  rowIndex={1}
  setValue={[MockFunction]}
  value={2.5}
  valueFormatted={3}
>
  <EnvironmentCellRenderer
    context={
      Object {
        "data": Object {
          "channelEnvironment": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelLag": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelMissing": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelTimeliness": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "id": "test",
          "needsAcknowledgement": false,
          "needsAttention": true,
          "stationData": Object {
            "stationCapabilityStatus": "GOOD",
            "stationName": "test",
            "stationStatus": "GOOD",
          },
          "stationEnvironment": 1,
          "stationGroups": Array [
            Object {
              "groupName": "Group",
              "sohStationCapability": "GOOD",
              "stationName": "test",
            },
          ],
          "stationLag": 2,
          "stationMissing": 3,
          "stationTimeliness": 4,
        },
      }
    }
    data={
      Object {
        "channelEnvironment": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelLag": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelMissing": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelTimeliness": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "id": "test",
        "needsAcknowledgement": false,
        "needsAttention": true,
        "stationData": Object {
          "stationCapabilityStatus": "GOOD",
          "stationName": "test",
          "stationStatus": "GOOD",
        },
        "stationEnvironment": 1,
        "stationGroups": Array [
          Object {
            "groupName": "Group",
            "sohStationCapability": "GOOD",
            "stationName": "test",
          },
        ],
        "stationLag": 2,
        "stationMissing": 3,
        "stationTimeliness": 4,
      }
    }
    formatValue={[MockFunction]}
    getValue={[MockFunction]}
    isStationCell={false}
    rowIndex={1}
    setValue={[MockFunction]}
    value={2.5}
    valueFormatted={3}
  >
    <StationStatisticsDragCell
      stationId="test"
    >
      <DragCell
        getSelectedStationIds={[Function]}
        getSingleDragImage={[Function]}
        setSelectedStationIds={[Function]}
        stationId="test"
      >
        <DragInitiator
          getDragImage={[Function]}
          getDragPayload={[Function]}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <div
            draggable={true}
            onDragEnd={[Function]}
            onDragStart={[Function]}
          >
            <div
              data-cy="environment-cell"
              title=""
            >
              <SohRollupCell
                cellStatus="good"
                className="table-cell--numeric"
                context={
                  Object {
                    "data": Object {
                      "channelEnvironment": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelLag": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelMissing": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelTimeliness": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "id": "test",
                      "needsAcknowledgement": false,
                      "needsAttention": true,
                      "stationData": Object {
                        "stationCapabilityStatus": "GOOD",
                        "stationName": "test",
                        "stationStatus": "GOOD",
                      },
                      "stationEnvironment": 1,
                      "stationGroups": Array [
                        Object {
                          "groupName": "Group",
                          "sohStationCapability": "GOOD",
                          "stationName": "test",
                        },
                      ],
                      "stationLag": 2,
                      "stationMissing": 3,
                      "stationTimeliness": 4,
                    },
                  }
                }
                data={
                  Object {
                    "channelEnvironment": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "channelLag": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "channelMissing": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "channelTimeliness": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "id": "test",
                    "needsAcknowledgement": false,
                    "needsAttention": true,
                    "stationData": Object {
                      "stationCapabilityStatus": "GOOD",
                      "stationName": "test",
                      "stationStatus": "GOOD",
                    },
                    "stationEnvironment": 1,
                    "stationGroups": Array [
                      Object {
                        "groupName": "Group",
                        "sohStationCapability": "GOOD",
                        "stationName": "test",
                      },
                    ],
                    "stationLag": 2,
                    "stationMissing": 3,
                    "stationTimeliness": 4,
                  }
                }
                dataReceivedStatus="received"
                formatValue={[MockFunction]}
                getValue={[MockFunction]}
                isStationCell={false}
                rowIndex={1}
                setValue={[MockFunction]}
                stationId="test"
                value="1.00"
                valueFormatted={3}
              >
                <TableCellRenderer
                  cellStatus="good"
                  className="soh-cell table-cell--numeric"
                  context={
                    Object {
                      "data": Object {
                        "channelEnvironment": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "channelLag": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "channelMissing": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "channelTimeliness": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "id": "test",
                        "needsAcknowledgement": false,
                        "needsAttention": true,
                        "stationData": Object {
                          "stationCapabilityStatus": "GOOD",
                          "stationName": "test",
                          "stationStatus": "GOOD",
                        },
                        "stationEnvironment": 1,
                        "stationGroups": Array [
                          Object {
                            "groupName": "Group",
                            "sohStationCapability": "GOOD",
                            "stationName": "test",
                          },
                        ],
                        "stationLag": 2,
                        "stationMissing": 3,
                        "stationTimeliness": 4,
                      },
                    }
                  }
                  data={
                    Object {
                      "channelEnvironment": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelLag": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelMissing": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelTimeliness": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "id": "test",
                      "needsAcknowledgement": false,
                      "needsAttention": true,
                      "stationData": Object {
                        "stationCapabilityStatus": "GOOD",
                        "stationName": "test",
                        "stationStatus": "GOOD",
                      },
                      "stationEnvironment": 1,
                      "stationGroups": Array [
                        Object {
                          "groupName": "Group",
                          "sohStationCapability": "GOOD",
                          "stationName": "test",
                        },
                      ],
                      "stationLag": 2,
                      "stationMissing": 3,
                      "stationTimeliness": 4,
                    }
                  }
                  data-cell-status="good"
                  data-received-status="received"
                  data-station-id="test"
                  dataReceivedStatus="received"
                  formatValue={[MockFunction]}
                  getValue={[MockFunction]}
                  isStationCell={false}
                  rowIndex={1}
                  setValue={[MockFunction]}
                  stationId="test"
                  value="1.00"
                  valueFormatted={3}
                >
                  <div
                    className="table-cell soh-cell table-cell--numeric"
                    data-cell-status="good"
                    data-cy="table-cell"
                    data-received-status="received"
                    data-station-id="test"
                    style={
                      Object {
                        "height": "36px",
                        "overflow": "hidden",
                      }
                    }
                    title="1.00"
                  >
                    <div
                      className=" table-cell__value"
                      data-cy="table-cell__value"
                    >
                      <span>
                        1.00
                      </span>
                    </div>
                    <PercentBar
                      percentage={1}
                    >
                      <div
                        className="percent-bar"
                        key="1"
                        style={
                          Object {
                            "transform": "translateX(-99%)",
                            "width": "100%",
                          }
                        }
                      />
                    </PercentBar>
                  </div>
                </TableCellRenderer>
              </SohRollupCell>
            </div>
          </div>
        </DragInitiator>
      </DragCell>
    </StationStatisticsDragCell>
  </EnvironmentCellRenderer>
</ChannelEnvironmentCellRenderer>
`;

exports[`Environment cell renderer StationEnvironmentCellRenderer can be created 1`] = `
<StationEnvironmentCellRenderer
  context={
    Object {
      "data": Object {
        "channelEnvironment": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelLag": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelMissing": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelTimeliness": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "id": "test",
        "needsAcknowledgement": false,
        "needsAttention": true,
        "stationData": Object {
          "stationCapabilityStatus": "GOOD",
          "stationName": "test",
          "stationStatus": "GOOD",
        },
        "stationEnvironment": 1,
        "stationGroups": Array [
          Object {
            "groupName": "Group",
            "sohStationCapability": "GOOD",
            "stationName": "test",
          },
        ],
        "stationLag": 2,
        "stationMissing": 3,
        "stationTimeliness": 4,
      },
    }
  }
  data={
    Object {
      "channelEnvironment": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelLag": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelMissing": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelTimeliness": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "id": "test",
      "needsAcknowledgement": false,
      "needsAttention": true,
      "stationData": Object {
        "stationCapabilityStatus": "GOOD",
        "stationName": "test",
        "stationStatus": "GOOD",
      },
      "stationEnvironment": 1,
      "stationGroups": Array [
        Object {
          "groupName": "Group",
          "sohStationCapability": "GOOD",
          "stationName": "test",
        },
      ],
      "stationLag": 2,
      "stationMissing": 3,
      "stationTimeliness": 4,
    }
  }
  formatValue={[MockFunction]}
  getValue={[MockFunction]}
  rowIndex={1}
  setValue={[MockFunction]}
  value={2.5}
  valueFormatted={3}
>
  <EnvironmentCellRenderer
    context={
      Object {
        "data": Object {
          "channelEnvironment": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelLag": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelMissing": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelTimeliness": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "id": "test",
          "needsAcknowledgement": false,
          "needsAttention": true,
          "stationData": Object {
            "stationCapabilityStatus": "GOOD",
            "stationName": "test",
            "stationStatus": "GOOD",
          },
          "stationEnvironment": 1,
          "stationGroups": Array [
            Object {
              "groupName": "Group",
              "sohStationCapability": "GOOD",
              "stationName": "test",
            },
          ],
          "stationLag": 2,
          "stationMissing": 3,
          "stationTimeliness": 4,
        },
      }
    }
    data={
      Object {
        "channelEnvironment": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelLag": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelMissing": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelTimeliness": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "id": "test",
        "needsAcknowledgement": false,
        "needsAttention": true,
        "stationData": Object {
          "stationCapabilityStatus": "GOOD",
          "stationName": "test",
          "stationStatus": "GOOD",
        },
        "stationEnvironment": 1,
        "stationGroups": Array [
          Object {
            "groupName": "Group",
            "sohStationCapability": "GOOD",
            "stationName": "test",
          },
        ],
        "stationLag": 2,
        "stationMissing": 3,
        "stationTimeliness": 4,
      }
    }
    formatValue={[MockFunction]}
    getValue={[MockFunction]}
    isStationCell={true}
    rowIndex={1}
    setValue={[MockFunction]}
    value={2.5}
    valueFormatted={3}
  >
    <StationStatisticsDragCell
      stationId="test"
    >
      <DragCell
        getSelectedStationIds={[Function]}
        getSingleDragImage={[Function]}
        setSelectedStationIds={[Function]}
        stationId="test"
      >
        <DragInitiator
          getDragImage={[Function]}
          getDragPayload={[Function]}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <div
            draggable={true}
            onDragEnd={[Function]}
            onDragStart={[Function]}
          >
            <div
              data-cy="environment-cell"
              title="1"
            >
              <SohRollupCell
                cellStatus="non-contributing"
                className="table-cell--numeric"
                context={
                  Object {
                    "data": Object {
                      "channelEnvironment": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelLag": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelMissing": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelTimeliness": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "id": "test",
                      "needsAcknowledgement": false,
                      "needsAttention": true,
                      "stationData": Object {
                        "stationCapabilityStatus": "GOOD",
                        "stationName": "test",
                        "stationStatus": "GOOD",
                      },
                      "stationEnvironment": 1,
                      "stationGroups": Array [
                        Object {
                          "groupName": "Group",
                          "sohStationCapability": "GOOD",
                          "stationName": "test",
                        },
                      ],
                      "stationLag": 2,
                      "stationMissing": 3,
                      "stationTimeliness": 4,
                    },
                  }
                }
                data={
                  Object {
                    "channelEnvironment": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "channelLag": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "channelMissing": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "channelTimeliness": Object {
                      "isContributing": true,
                      "status": "GOOD",
                      "value": 1,
                    },
                    "id": "test",
                    "needsAcknowledgement": false,
                    "needsAttention": true,
                    "stationData": Object {
                      "stationCapabilityStatus": "GOOD",
                      "stationName": "test",
                      "stationStatus": "GOOD",
                    },
                    "stationEnvironment": 1,
                    "stationGroups": Array [
                      Object {
                        "groupName": "Group",
                        "sohStationCapability": "GOOD",
                        "stationName": "test",
                      },
                    ],
                    "stationLag": 2,
                    "stationMissing": 3,
                    "stationTimeliness": 4,
                  }
                }
                dataReceivedStatus="received"
                formatValue={[MockFunction]}
                getValue={[MockFunction]}
                isStationCell={true}
                rowIndex={1}
                setValue={[MockFunction]}
                stationId="test"
                value="1.00"
                valueFormatted={3}
              >
                <TableCellRenderer
                  cellStatus="non-contributing"
                  className="soh-cell table-cell--numeric"
                  context={
                    Object {
                      "data": Object {
                        "channelEnvironment": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "channelLag": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "channelMissing": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "channelTimeliness": Object {
                          "isContributing": true,
                          "status": "GOOD",
                          "value": 1,
                        },
                        "id": "test",
                        "needsAcknowledgement": false,
                        "needsAttention": true,
                        "stationData": Object {
                          "stationCapabilityStatus": "GOOD",
                          "stationName": "test",
                          "stationStatus": "GOOD",
                        },
                        "stationEnvironment": 1,
                        "stationGroups": Array [
                          Object {
                            "groupName": "Group",
                            "sohStationCapability": "GOOD",
                            "stationName": "test",
                          },
                        ],
                        "stationLag": 2,
                        "stationMissing": 3,
                        "stationTimeliness": 4,
                      },
                    }
                  }
                  data={
                    Object {
                      "channelEnvironment": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelLag": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelMissing": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "channelTimeliness": Object {
                        "isContributing": true,
                        "status": "GOOD",
                        "value": 1,
                      },
                      "id": "test",
                      "needsAcknowledgement": false,
                      "needsAttention": true,
                      "stationData": Object {
                        "stationCapabilityStatus": "GOOD",
                        "stationName": "test",
                        "stationStatus": "GOOD",
                      },
                      "stationEnvironment": 1,
                      "stationGroups": Array [
                        Object {
                          "groupName": "Group",
                          "sohStationCapability": "GOOD",
                          "stationName": "test",
                        },
                      ],
                      "stationLag": 2,
                      "stationMissing": 3,
                      "stationTimeliness": 4,
                    }
                  }
                  data-cell-status="non-contributing"
                  data-received-status="received"
                  data-station-id="test"
                  dataReceivedStatus="received"
                  formatValue={[MockFunction]}
                  getValue={[MockFunction]}
                  isStationCell={true}
                  rowIndex={1}
                  setValue={[MockFunction]}
                  stationId="test"
                  value="1.00"
                  valueFormatted={3}
                >
                  <div
                    className="table-cell soh-cell table-cell--numeric"
                    data-cell-status="non-contributing"
                    data-cy="table-cell"
                    data-received-status="received"
                    data-station-id="test"
                    style={
                      Object {
                        "height": "36px",
                        "overflow": "hidden",
                      }
                    }
                    title="1.00"
                  >
                    <div
                      className=" table-cell__value"
                      data-cy="table-cell__value"
                    >
                      <span>
                        1.00
                      </span>
                    </div>
                    <PercentBar
                      percentage={1}
                    >
                      <div
                        className="percent-bar"
                        key="2"
                        style={
                          Object {
                            "transform": "translateX(-99%)",
                            "width": "100%",
                          }
                        }
                      />
                    </PercentBar>
                  </div>
                </TableCellRenderer>
              </SohRollupCell>
            </div>
          </div>
        </DragInitiator>
      </DragCell>
    </StationStatisticsDragCell>
  </EnvironmentCellRenderer>
</StationEnvironmentCellRenderer>
`;

exports[`Environment cell renderer can be created handle no data 1`] = `
<StationEnvironmentCellRenderer
  context={
    Object {
      "data": Object {
        "channelEnvironment": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelLag": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelMissing": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelTimeliness": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "id": "test",
        "needsAcknowledgement": false,
        "needsAttention": true,
        "stationData": Object {
          "stationCapabilityStatus": "GOOD",
          "stationName": "test",
          "stationStatus": "GOOD",
        },
        "stationEnvironment": 1,
        "stationGroups": Array [
          Object {
            "groupName": "Group",
            "sohStationCapability": "GOOD",
            "stationName": "test",
          },
        ],
        "stationLag": 2,
        "stationMissing": 3,
        "stationTimeliness": 4,
      },
    }
  }
  data={
    Object {
      "channelEnvironment": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelLag": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelMissing": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "channelTimeliness": Object {
        "isContributing": true,
        "status": "GOOD",
        "value": 1,
      },
      "id": "test",
      "needsAcknowledgement": false,
      "needsAttention": true,
      "stationData": Object {
        "stationCapabilityStatus": "GOOD",
        "stationName": "test",
        "stationStatus": "GOOD",
      },
      "stationEnvironment": 1,
      "stationGroups": Array [
        Object {
          "groupName": "Group",
          "sohStationCapability": "GOOD",
          "stationName": "test",
        },
      ],
      "stationLag": 2,
      "stationMissing": 3,
      "stationTimeliness": 4,
    }
  }
  formatValue={[MockFunction]}
  getValue={[MockFunction]}
  rowIndex={1}
  setValue={[MockFunction]}
  value={2.5}
  valueFormatted={3}
>
  <EnvironmentCellRenderer
    context={
      Object {
        "data": Object {
          "channelEnvironment": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelLag": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelMissing": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "channelTimeliness": Object {
            "isContributing": true,
            "status": "GOOD",
            "value": 1,
          },
          "id": "test",
          "needsAcknowledgement": false,
          "needsAttention": true,
          "stationData": Object {
            "stationCapabilityStatus": "GOOD",
            "stationName": "test",
            "stationStatus": "GOOD",
          },
          "stationEnvironment": 1,
          "stationGroups": Array [
            Object {
              "groupName": "Group",
              "sohStationCapability": "GOOD",
              "stationName": "test",
            },
          ],
          "stationLag": 2,
          "stationMissing": 3,
          "stationTimeliness": 4,
        },
      }
    }
    data={
      Object {
        "channelEnvironment": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelLag": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelMissing": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "channelTimeliness": Object {
          "isContributing": true,
          "status": "GOOD",
          "value": 1,
        },
        "id": "test",
        "needsAcknowledgement": false,
        "needsAttention": true,
        "stationData": Object {
          "stationCapabilityStatus": "GOOD",
          "stationName": "test",
          "stationStatus": "GOOD",
        },
        "stationEnvironment": 1,
        "stationGroups": Array [
          Object {
            "groupName": "Group",
            "sohStationCapability": "GOOD",
            "stationName": "test",
          },
        ],
        "stationLag": 2,
        "stationMissing": 3,
        "stationTimeliness": 4,
      }
    }
    formatValue={[MockFunction]}
    getValue={[MockFunction]}
    isStationCell={true}
    rowIndex={1}
    setValue={[MockFunction]}
    value={2.5}
    valueFormatted={3}
  />
</StationEnvironmentCellRenderer>
`;
